worker/agent-server.ts

```
@@ -264,6 +264,40 @@ export class AgentServer {
       }
     });
 
+    this.app.post(
+      "/api/reboot",
+      authMiddleware,
+      async (req: express.Request, res: express.Response) => {
+        try {
+          this.logger.info("Reboot endpoint called");
+          const { rebootAgent } = await import("./reboot-manager");
+          const success = await rebootAgent(undefined, this.logger);
+
+          if (success) {
+            res.json({
+              status: "success",
+              message: "Agent reboot initiated",
+              timestamp: new Date().toISOString(),
+            });
+          } else {
+            res.status(500).json({
+              status: "error",
+              message: "Failed to initiate reboot",
```

If success is false, rebootAgent should return the failure reason too